weaviate/collection/collection.py

```diff
@@ -1,67 +1,86 @@
-from typing import Optional
+from typing import Generic, Optional, Type
 
 from weaviate.collection.classes.config import CollectionConfig
+from weaviate.collection.classes.internal import Properties
 from weaviate.collection.collection_base import CollectionBase
 from weaviate.collection.config import _ConfigCollection
 from weaviate.collection.data import _DataCollection
-from weaviate.collection.grpc import _GrpcCollection, _RawObject
+from weaviate.collection.grpc import _GrpcCollection
 from weaviate.collection.tenants import _Tenants
 from weaviate.connect import Connection
 from weaviate.data.replication import ConsistencyLevel
 from weaviate.util import _capitalize_first_letter
 
 
-class CollectionObject:
+class CollectionObject(Generic[Properties]):
     def __init__(
         self,
         connection: Connection,
         name: str,
-        config: _ConfigCollection,
+        type_: Optional[Type[Properties]] = None,
```

Comment:
can we rename it to data_model everywhere for consistency?

Reply:
I think `type_` makes more sense than `data_model` in the internal context with `data_model` just acting as the user-preferred nomenclature

Since Python calls its module `typing`, the type hint of `type_` is `Type[...]`, within `_DataCollection` the `type_` argument is used in a `get_type_hints` function call

I also imagine calling this `data_model` might become confusing when working with the ORM where it will either be just `model` or also `data_model`, which would be confusing in my book. Calling it `type_` here makes it clear that it is only a type hint and not an ORM model